Microsoft.Toolkit.HighPerformance/Enumerables/ReadOnlyRefEnumerable{T}.cs

```diff
@@ -116,21 +155,53 @@ internal ReadOnlyRefEnumerable(object? instance, IntPtr offset, int length, int
         {
             this.instance = instance;
             this.offset = offset;
-            this.length = length;
+            Length = length;
             this.step = step;
         }
 #endif
 
+        /// <summary>
+        /// Returns a reference to the first element within the current instance, with no bounds check.
+        /// </summary>
+        /// <returns>A reference to the first element within the current instance.</returns>
+        internal ref readonly T DangerousGetReference()
+        {
+#if SPAN_RUNTIME_SUPPORT
+            return ref MemoryMarshal.GetReference(this.span);
+#else
+            return ref RuntimeHelpers.GetObjectDataAtOffsetOrPointerReference<T>(this.instance, this.offset);
+#endif
+        }
+
+        /// <summary>
+        /// Returns a reference to a specified element within the current instance, with no bounds check.
+        /// </summary>
+        /// <returns>A reference to the element at the specified index.</returns>
+        internal ref readonly T DangerousGetReferenceAt(int index)
+        {
+#if SPAN_RUNTIME_SUPPORT
+            ref T r0 = ref MemoryMarshal.GetReference(this.span);
+#else
+            ref T r0 = ref RuntimeHelpers.GetObjectDataAtOffsetOrPointerReference<T>(this.instance, this.offset);
+#endif
+
+            // Here we just offset by shifting down as if we were traversing a 2D array with a
+            // a single column, with the width of each row represented by the step, the height
+            // represented by the current position, and with only the first element of each row
+            // being inspected. We can perform all the indexing operations in this type as nint,
+            // as the maximum offset is guaranteed never to exceed the maximum value, since on
+            // 32 bit architectures it's not possible to allocate that much memory anyway.
```

Comment:
You forgot to remove or at least update the comments when porting this bit.
You should just remove these comments since they're duplicated anyway.